### PR TITLE
multi-selectable job rows w/ shift select, actions bar

### LINKED ISSUE #54: Select and cancel/delete/retry from job list
I pushed a bad commit to the demo this morning and had a bunch of jobs to cancel and delete. Doing this without a batch selection UI was abysmal, so this is clearly needed.

### PR DESCRIPTION
**Pull request by bgentry**
This adds checkboxes to each row on the job list, as well as a select/deselect all box on a new header row above the list. Checking any jobs causes the job list to stop refreshing until everything is unchecked.

Depending on the state being viewed, selected jobs can be retried, cancelled, or deleted as a batch. Performing any action on selected jobs causes the TanStack query cache to be invalidated for the job list and the counts by state so those get immediately refetched.

Selection even works with the shift key to select all jobs between the previous one you checked and the one you just checked.

localhost_5173_jobs_state=running
localhost_5173_jobs_state=running (1)
localhost_5173_jobs_state=running (2)

Fixes #54.

### CODE REVIEW

**Comment by bgentry** [comment]
@brandur I'm gonna get this shipped to the demo and we can fix any remaining issues as follow-ups after testing it out more there. The basics all seem to be working great in my local testing 🚀 :shipit:

File: ui/src/components/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  "dark:after:-inset-px dark:after:hidden dark:after:rounded-[0.3125rem] dark:group-data-[checked]:after:block",
+  // Focus ring
+  "group-data-[focus]:outline group-data-[focus]:outline-2 group-data-[focus]:outline-offset-2 group-data-[focus]:outline-blue-500",
+  // Disabled state
+  "group-data-[disabled]:opacity-50",
+  "group-data-[disabled]:border-zinc-950/25 group-data-[disabled]:bg-zinc-950/5 group-data-[disabled]:[--checkbox-check:theme(colors.zinc.950/50%)] group-data-[disabled]:before:bg-transparent",
+  "dark:group-data-[disabled]:border-white/20 dark:group-data-[disabled]:bg-white/[2.5%] dark:group-data-[disabled]:[--checkbox-check:theme(colors.white/50%)] dark:group-data-[disabled]:group-data-[checked]:after:hidden",
+  // Forced colors mode
+  "forced-colors:[--checkbox-check:HighlightText] forced-colors:[--checkbox-checked-bg:Highlight] forced-colors:group-data-[disabled]:[--checkbox-check:Highlight]",
+  "dark:forced-colors:[--checkbox-check:HighlightText] dark:forced-colors:[--checkbox-checked-bg:Highlight] dark:forced-colors:group-data-[disabled]:[--checkbox-check:Highlight]",
```
Comment (brandur) [inline ui/src/components/Checkbox.tsx:78]: Hm, okay so I know JS is generally prone to this sort of stuff, but doesn't this strike you as just _a lot_ of code to style a checkbox? IMO, it does kind of matter because if you ever need to change something, it's fairly hard to parse what's going on and what needs to be changed?

File: ui/src/components/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  indigo:
+    "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.indigo.500)] [--checkbox-checked-border:theme(colors.indigo.600/90%)]",
+  violet:
+    "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.violet.500)] [--checkbox-checked-border:theme(colors.violet.600/90%)]",
+  purple:
+    "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.purple.500)] [--checkbox-checked-border:theme(colors.purple.600/90%)]",
+  fuchsia:
+    "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.fuchsia.500)] [--checkbox-checked-border:theme(colors.fuchsia.600/90%)]",
+  pink: "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.pink.500)] [--checkbox-checked-border:theme(colors.pink.600/90%)]",
+  rose: "[--checkbox-check:theme(colors.white)] [--checkbox-checked-bg:theme(colors.rose.500)] [--checkbox-checked-border:theme(colors.rose.600/90%)]",
```
Comment (brandur) [inline ui/src/components/Checkbox.tsx:119]: Similarly, it's necessary to reinitialize every Tailwind color like this?

**Review by brandur — COMMENTED** [review]
Awesome.
